### PR TITLE
 msys2 setup step removed

### LINKED ISSUE #12: Remove `Setup-Msys2` action
& use `pacman` in a way that ensures all of the necessary tools are installed but doesn't download and re-install things that are already installed and up-to-date.

### PR DESCRIPTION
**Pull request by ilkerhalil**
- Fixed Linux header artifact location
- Changed Windows build strategy

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -9,28 +9,23 @@ jobs:
...
     - name: Add msbuild to PATH
       uses: microsoft/setup-msbuild@v1.0.2
+    - name: Install msys2 dependencies
+      run: |
+        c:\msys64\msys2_shell.cmd -defterm -no-start -here -mingw64 -c 'pacman -S --noconfirm libxslt'
     - name: Build Generic64
-      shell: msys2 {0}
       run: |
-        CFLAGS="-O3 -g0 -march=ivybridge -mtune=native" ASMFLAGS="-march=ivybridge -mtune=native" make generic64/libXKCP.so
+        c:\msys64\msys2_shell.cmd -defterm -no-start -here -mingw64 -c "CFLAGS='-O3 -g0 -march=ivybridge -mtune=native' ASMFLAGS='-march=ivybridge -mtune=native' make generic64/libXKCP.so"
```
Comment (Rjvs) [inline .github/workflows/release.yml]: Does something like this not work?  I removed the shell.cmd and args which you might need but there are definitely some improvements in readability possible.

```suggestion
        make generic64/libXKCP.so
      env:
        CFLAGS: "-O3 -g0 -march=ivybridge -mtune=native"
        ASMFLAGS: "-march=ivybridge -mtune=native"
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -9,28 +9,23 @@ jobs:
...
     - uses: actions/checkout@v2
-    - uses: msys2/setup-msys2@v2
-    - name: Install dependencies
-      run: |
-         msys2 -c 'pacman -S --noconfirm libxslt base-devel'
     - name: Add msbuild to PATH
       uses: microsoft/setup-msbuild@v1.0.2
+    - name: Install msys2 dependencies
+      run: |
+        c:\msys64\msys2_shell.cmd -defterm -no-start -here -mingw64 -c 'pacman -S --noconfirm libxslt'
```
Comment (Rjvs) [inline .github/workflows/release.yml]: I don't have a run of this on the SL repo to check but is it still downloading and reinstalling heaps of dependencies?

File: .github/workflows/release.yml (hunk 1 above)
Comment (Rjvs) [reply .github/workflows/release.yml]: Note that you can get more readability by setting the environment variables above the run command.

**Review by Rjvs — CHANGES_REQUESTED** [review]
I think we can make this more readable.

File: .github/workflows/release.yml (hunk 2 above)
Comment (TYoungSL) [reply .github/workflows/release.yml]: I don't think that will.

File: .github/workflows/release.yml (hunk 1 above)
Comment (TYoungSL) [reply .github/workflows/release.yml]: It should work.

File: .github/workflows/release.yml (hunk 2 above)
Comment (TYoungSL) [reply .github/workflows/release.yml]: Ran it locally, seems to just update libxslt.

File: .github/workflows/release.yml (hunk 1 above)
Comment (Rjvs) [reply .github/workflows/release.yml]: @ilkerhalil ?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -17,20 +17,26 @@ jobs:
...
       shell: msys2_shell.cmd {0}
       run: |
-          -defterm -no-start -here -mingw64 -c 'pacman -S --noconfirm libxslt'
-
+        -defterm -no-start -here -mingw64 -c 'pacman -S --noconfirm libxslt'
     - name: Build Generic64
+      shell: msys2_shell.cmd {0}
       run: |
-        c:\msys64\msys2_shell.cmd -defterm -no-start -here -mingw64 -c "CFLAGS='-O3 -g0 -march=ivybridge -mtune=native' ASMFLAGS='-march=ivybridge -mtune=native' make generic64/libXKCP.so"
+        -defterm -no-start -here -mingw64 -c "make generic64/libXKCP.so"
```
Comment (TYoungSL) [inline .github/workflows/release.yml]: Maybe try this...
```suggestion
      shell: msys2_shell.cmd -defterm -no-start -here -mingw64 {0}
      run: |
         make generic64/libXKCP.so
```
I think it's going to replace `{0}` with a file it writes the script in `run:` to, so drop the `-c`.

**Review by TYoungSL — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (ilkerhalil) [reply .github/workflows/release.yml]: @Rjvs It worked as Tyler suggested.

**Review by Rjvs — COMMENTED** [review]
LGTM

**Pull request merged**